### PR TITLE
Doc indexing

### PR DESCRIPTION
**Pull request by feipenghe**
This PR creates a indexing page for tutorials.

### Description of changes
* create a indexing page
* move tutorials into folders under `tutorial` folder

### Possible influences of this PR.
Current we don't have a structured tutorial indexing page, and it's hard for beginners to find tutorial they want.
This PR creates a indexing page helping them find tutorials suit users' needs.

### Test Conducted
No test needed.

### CODE REVIEW

File: docs/outline.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  - [Text Classification](#text-classification)
+- [4. Linguistic Analysis](#4-linguistic-analysis)
+  - [Part-of-Speech Tagging](#part-of-speech-tagging)
+- [5. Language Modeling with External Library](#5-language-modeling-with-external-library)
+- [6. Application](#6-application)
+  - [1. Information Extraction](#1-information-extraction)
+    - [Named Entity Recognition](#named-entity-recognition)
+    - [Relation Extraction](#relation-extraction)
+    - [Indexing](#indexing)
+    - [Ranking](#ranking)
```
Comment (hunterhector) [inline docs/outline.md:18]: indexing and ranking are information retrieval

File: docs/outline.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  - [2. Machine Translation](#2-machine-translation)
+  - [3. Text Generation](#3-text-generation)
+  - [4. Question Answering System](#4-question-answering-system)
+  - [5. Chatbot](#5-chatbot)
+- [7. More data format](#7-more-data-format)
```
Comment (hunterhector) [inline docs/outline.md:23]: Consider using tasks (e.g. ASR)

File: docs/outline.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+- [0. Overview](#0-overview)
+- [1. Handling Structured Text Data](#1-handling-structured-text-data)
+- [2. Processing Raw Text](#2-processing-raw-text)
+  - [Reader](#reader)
+  - [Tokenizer](#tokenizer)
```
Comment (hunterhector) [inline docs/outline.md:5]: Tokenization, to be fair, is a task by itself.

File: docs/outline.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    - [Relation Extraction](#relation-extraction)
+    - [Indexing](#indexing)
+    - [Ranking](#ranking)
+  - [2. Machine Translation](#2-machine-translation)
+  - [3. Text Generation](#3-text-generation)
+  - [4. Question Answering System](#4-question-answering-system)
+  - [5. Chatbot](#5-chatbot)
+- [7. More data format](#7-more-data-format)
+  - [Audio Data](#audio-data)
+  - [Image Data](#image-data)
```
Comment (hunterhector) [inline docs/outline.md:25]: You probably don't need a TOC since sphinx will have a sidebar. but the TOC is good

Comment (feipenghe) [reply docs/outline.md:30]: ok I think we can leave it there and delete it later if it's not useful.

**Pull request merged**